cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1271,6 +1271,57 @@ def _check_column_availability(self):
                 if "add_labels" in index_def:
                     self._check_single_column_availability(component, index_def["add_labels"])
 
+    def _validate_dense_raw_matrix_values(self) -> None:
+        raw_matrix = self._get_raw_x()
+        for row_index in range(0, self.adata.n_obs):
+            row = raw_matrix[row_index]
+            whole_row_is_zeros = True
+            for col_index, val in enumerate(row):
+                if val:
+                    whole_row_is_zeros = False
+                if not isinstance(val, np.float32) or val != val.astype(int):
+                    self.errors.append(
+                        f"All non-zero values must be positive integers converted to the numpy.float32 type. "
+                        f"The value at [{row_index}, {col_index}] does not match this requirement."
+                    )
+                    break
+            if whole_row_is_zeros:
+                self.errors.append(
+                    f"Each cell must have at least one non-zero value in its row. The row at index {row_index} "
+                    f"fails this requirement."
+                )
+                break
+
+    def _validate_sparse_raw_matrix_values(self) -> None:
+        rows_without_nonzero = {*range(0, self.adata.n_obs)}
+        row_indices, col_indices = self._get_raw_x().nonzero()
+        for row_index in row_indices:
+            if row_index in rows_without_nonzero:
+                rows_without_nonzero.remove(row_index)
+        if rows_without_nonzero:
+            self.errors.append(
+                f"Each cell must have at least one non-zero value in its row. The full set of "
+                f"rows which fail this requirement: {rows_without_nonzero}"
+            )
+        for row_index, col_index in zip(row_indices, col_indices):
+            val = self._get_raw_x()[row_index, col_index]
+            if not isinstance(val, np.float32) or val != val.astype(int):
+                self.errors.append(
+                    f"All non-zero values must be positive integers converted to the numpy.float32 type. The value"
+                    f"at [{row_index}, {col_index}] does not match this requirement."
+                )
+                break
+
+    def _check_raw_matrix_values(self):
+        """
+        Validate that each row of the raw matrix has at least one nonzero value and that all values are numpy.float32
+        """
+        matrix_format = get_matrix_format()
+        if matrix_format in ("csr", "csc", "coo"):  # full set of sparse matrix types
+            self._validate_sparse_raw_matrix_values()
+        elif matrix_format == "dense":
+            self._validate_dense_raw_matrix_values()
```

Comment:
Have not considered yet what should happen in cases where matrix_format is "unknown".